astro/src/content/articles/authentication/bff.mdx

```diff
@@ -0,0 +1,319 @@
+
+- For a web app, a BFF is highly valuable: It allows the app to store nothing but a cookie, which attackers can't exfiltrate.
+- For a mobile or desktop app, a BFF is less valuable because these apps may not store cookies like browsers do. However, a BFF can still return a short-lived authentication credential, so even if it's stolen, the attacker has limited time to use it. Without a BFF, a stolen refresh token could allow an attacker to impersonate a user for days.
+
+A BFF may also expose fewer endpoints to the frontend, limiting the actions an attacker can take, and detect unusual behavior – blocking suspicious requests before they reach the business server, much like banks check for credit card fraud.
+
+Read more about [where to store tokens on different devices](/articles/oauth/oauth-token-storage).
+
+## OAuth Allows Serverless Apps, But Don't Use Them
+
+In his 2023 [talk on the non-security of serverless OAuth apps](https://www.youtube.com/watch?v=OpFN6gmct8c), Philippe de Ryck noted that the OAuth organization doesn't discourage programmers from using serverless apps strongly enough. OAuth recommends:
+
+- Using PKCE, which you've seen still allows theft of refresh tokens.
+- Changing refresh tokens frequently, which doesn't prevent an attacker from waiting until a user closes the website and then using the latest refresh token to impersonate the user for hours or days.
+- Hiding credentials in service workers in the browser, as service workers are less vulnerable to XSS attacks. But service workers are started by JavaScript and so can be stopped by malicious JavaScript.
+
+Philippe's conclusion is that there is ultimately no way to adequately protect a serverless OAuth frontend.
+
+## How To Create A BFF For A Serverless App
+
+Let's look at some code to create a BFF for a currently serverless app – first to see how the app runs and login works, then to see how the code works.
+
+Clone or download the repository from https://github.com/fusionauth/fusionauth-example-hostedbackend. 
+
+You will need Docker installed to run the example.
+
+In this example, you'll use the authentication gateway FusionAuth as your OAuth provider. FusionAuth is free for basic usage with unlimited users and applications, and can function either as an identity provider or as a way to connect to identity providers like GitHub and Google.
+
+Open a terminal in the project directory you downloaded and run the command below to start FusionAuth.
+
+```sh
+docker compose up
+```
+
+Wait a minute for `Kickstarting 🤘` to finish, then browse to http://localhost:9011 and log in with `admin@example.com` and `password`. You can browse around FusionAuth to see the example application and users.
+
+In a new terminal, run the commands below to start the Node web app.
+
+```sh
+cd nodeApp
+
+docker run --init -it --rm --name "app" -v ".:/app" -w "/app" -p 3000:3000 --network faNetwork node:23-alpine3.19 sh -c  "npm install && node app.js"
```

Comment:
I wasn't able to get the sample up and running due to a Docker error saying it couldn't connect to the network. I'm not sure if this is an issue on my end, or if it's something in the config. I've asked teammates to verify for me on Monday.